sklearn/decomposition/_nmf.py

```diff
@@ -1264,6 +1221,48 @@ def __init__(self, n_components=None, *, init='warn', solver='cd',
     def _more_tags(self):
         return {'requires_positive_X': True}
 
+    def _check_params(self, n_components, max_iter, tol):
```

Comment:
Now that it's a method we can use the attributes instead of passing them as arguments.